packages/components/src/data-model/word-query/word-query-resolver.js

```diff
@@ -0,0 +1,79 @@
+import { makeVar } from '@apollo/client/core'
+import { Logger, Digest, Language } from 'alpheios-data-models'
+import WordQuery from '@comp/data-model/word-query/word-query.js'
+import WordQueryResponse from '@comp/data-model/word-query/word-query-response.js'
+
+/**
+ * A class that would receive GraphQL requests and resolve them with data.
+ * Creates an instance of WordQuery for the query resolution.
+ */
+export default class WordQueryResolver {
+  constructor ({ getLexiconsFn, getShortLexiconsFn } = {}) {
+    this._getLexiconsFn = getLexiconsFn
+    this._getShortLexiconsFn = getShortLexiconsFn
+  }
+
+  /**
+   * This method is called when a new GraphQL query arrives.
+   *
+   * @param {*} _ - A parameter that is not used in a method, underscore is used to designate
+   *                such a parameter, according to Apollo convention.
+   * @param {object} options - An options object.
+   * @param {object} options.storage - A storage object that outlives the query.
+   * @param {object} options.variables - Parameters of a GraphQL query.
+   * @returns {object} - A result of a GraphQL query.
+   */
+  resolve (_, { storage, variables }) {
+    // This is a unique key of each request. We'll use its hash value to cache requests already resolved.
+    const key = Digest.fromObject({
+      word: variables.word,
+      language: variables.language,
+      contextForward: variables.contextForward,
+      contextBackward: variables.contextBackward
+    })
+    // The words map will store references to the results stored in the Apollo cache
+    if (!storage.words) { storage.words = new Map() }
+    if (!storage.words.has(key)) {
+      // eslint-disable-next-line prefer-const
+      let response = new WordQueryResponse({ updateCallback: this.updateCallback.bind(this, storage, key) })
+      const lexicons = this._getLexiconsFn(variables.language, variables.location)
+      const shortLexicons = this._getShortLexiconsFn(variables.language, variables.location)
+      const wordQuery = new WordQuery({
+        word: variables.word,
+        language: Language.fromJsonObject({ code: variables.language }),
+        clientId: 'graphql-client',
```

Comment:
I think the client Id should be more specific -- can we construct it from the application build info?

Reply:
Yes, we definitely could do that. What's the purpose of the client ID on the server side? How detailed should it be? As detailed as possible?

Reply:
it's used to track usage in the web server logs. So yes the more detailed the better.

Reply:
Thanks for explaining, will make this change